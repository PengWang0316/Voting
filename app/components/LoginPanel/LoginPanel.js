import React from 'react';
import { View } from 'react-native';

import LoginForm from '../LoginForm';
import styles from './Styles';

type Props = { navigation: Object };

export const LoginPanel = ({ handleSnackbarUpdat, navigation }: Props) => (
  <View style={styles.internalContainer}>
    <LoginForm handleSnackbarUpdat={handleSnackbarUpdat} navigation={navigation} />
  </View>
);
export default LoginPanel;
