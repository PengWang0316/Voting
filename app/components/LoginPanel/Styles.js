import { StyleSheet } from 'react-native';

import Theme from '../../Theme';

export default StyleSheet.create({
  container: {
    flex: 1,
    justifyContent: 'center',
    alignItems: 'center',
    backgroundColor: Theme.panelBackgroundColor,
  },
  internalContainer: {
    width: '75%',
  },
  deviderView: {
    flex: 1,
    justifyContent: 'space-around',
    alignItems: 'center',
    flexDirection: 'row',
    marginTop: 30,
    marginBottom: 30,
  },
  devider: {
    backgroundColor: Theme.primary.main,
    height: 1,
    width: '38%',
  },
});
