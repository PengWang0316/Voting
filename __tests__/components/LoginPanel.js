import React from 'react';
import renderer from 'react-test-renderer';

import { LoginPanel } from '../../app/components/LoginPanel/LoginPanel';

jest.mock('../../app/components/LoginForm', () => 'LoginForm');

describe('LoginPanel', () => {
  const defaultProps = { handleSnackbarUpdate: jest.fn() };
  test('Snapshot', () => expect(renderer.create(<LoginPanel {...defaultProps} />).toJSON()).toMatchSnapshot());
});
