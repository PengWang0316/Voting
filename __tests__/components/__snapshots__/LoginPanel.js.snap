// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`LoginPanel Snapshot 1`] = `
<View
  style={
    Object {
      "width": "75%",
    }
  }
>
  <LoginForm />
</View>
`;
